### PR TITLE
python3Packages.smartypants: fix src url

### LINKED ISSUE #118732: smartypants source doesn’t download any more
```
nix-build -A pkgs.python3Packages.smartypants
```
depends on 
```
  src = fetchhg {
    url = "https://bitbucket.org/livibetter/smartypants.py";
    rev = "v${version}";
    sha256 = "1cmzz44d2hm6y8jj2xcq1wfr26760gi7iq92ha8xbhb1axzd7nq6";
  };
```
but that doesn’t download any more:
```
~/build/nixpkgs $ nix-build /nix/store/3wlzl4h3vqqkb5hvgzmbgqx0b5p9rinp-hg-archive.drv --substituters ''
these derivations will be built:
  /nix/store/3wlzl4h3vqqkb5hvgzmbgqx0b5p9rinp-hg-archive.drv
building '/nix/store/3wlzl4h3vqqkb5hvgzmbgqx0b5p9rinp-hg-archive.drv'...
getting https://bitbucket.org/livibetter/smartypants.py (v1.8.6) into /nix/store/cvxh8lbxs0v0q28amldxbmpaky87mnnl-hg-archive
warning: connection security to bitbucket.org is disabled per current settings; communication is susceptible to eavesdropping and tampering
abort: HTTP Error 404: Not Found
builder for '/nix/store/3wlzl4h3vqqkb5hvgzmbgqx0b5p9rinp-hg-archive.drv' failed with exit code 255
error: build of '/nix/store/3wlzl4h3vqqkb5hvgzmbgqx0b5p9rinp-hg-archive.drv' failed
```

@costrouc  maybe?

### PR DESCRIPTION
**Pull request by veprbl**
###### Motivation for this change

Close: #118732

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/smartypants/default.nix [hunk 1]
```diff
@@ -9,10 +9,10 @@ buildPythonPackage rec {
   pname = "smartypants";
   disabled = isPyPy;
 
-  src = fetchhg {
-    url = "https://bitbucket.org/livibetter/smartypants.py";
+  src = fetchgit {
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/smartypants/default.nix:12]: Please use fetchFromGitHub.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 118795` at dc8e2d3e run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>645 packages marked as broken and skipped:</summary>
  <ul>
    <li>adoptopenjdk-hotspot-bin-13</li>
    <li>adoptopenjdk-hotspot-bin-14</li>
    <li>adoptopenjdk-jre-hotspot-bin-13</li>
    <li>adoptopenjdk-jre-hotspot-bin-14</li>
    <li>agdaPackages.iowa-stdlib</li>
    <li>aldor</li>
    <li>ams-lv2</li>
    <li>amule</li>
    <li>amuleDaemon</li>
    <li>amuleGui</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>5983 packages skipped due to time constraints:</summary>
  <ul>
    <li>AusweisApp2</li>
    <li>DisnixWebService</li>
    <li>Fabric (python38Packages.Fabric)</li>
    <li>MIDIVisualizer</li>
    <li>R</li>
    <li>SDL_Pango</li>
    <li>SDL_mixer</li>
    <li>abcl</li>
    <li>abcm2ps</li>
    <li>abiword</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>9 packages built successfully:</summary>
  <ul>
    <li>asciidoctor</li>
    <li>gi-docgen</li>
    <li>git</li>
    <li>pango (gnome2.pango)</li>
    <li>python38Packages.smartypants</li>
    <li>python38Packages.typogrify</li>
    <li>python39Packages.pelican</li>
    <li>python39Packages.smartypants</li>
    <li>python39Packages.typogrify</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/development/python-modules/smartypants/default.nix:23:5:
```
   |
23 |     maintainers = with maintainers; [ ];
   |     ^
```
</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 118795` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>python38Packages.smartypants</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
Reverted in #118814 and retargeting to staging.

**Pull request merged**